src/HtmlParser/Util.elm

```diff
@@ -390,3 +392,279 @@ toVirtualDomEach node =
 toAttribute : (String, String) -> Attribute msg
 toAttribute (name, value) =
   attribute name value
+
+{-| Converts nodes to virtual dom SVG nodes
+-}
+toVirtualDomSvg : List HtmlParser.Node -> List (Svg msg)
+toVirtualDomSvg nodes =
+  List.map toVirtualDomSvgEach nodes
+
+toVirtualDomSvgEach : HtmlParser.Node -> Svg msg
+toVirtualDomSvgEach node =
+  case node of
+    HtmlParser.Element name attrs children ->
+      Svg.node name (List.filterMap toSvgAttribute attrs) (toVirtualDomSvg children)
+
+    HtmlParser.Text s ->
+      text s
+
+    HtmlParser.Comment _ ->
+      text ""
+
+toSvgAttribute : (String, String) -> Maybe (Attribute msg)
+toSvgAttribute (name, value) =
+  case name of
```

Comment:
You can move the `Just <|` up above the `case` and let it be focused on creating the value instead of worrying about the API of the function.